app/views/portal/show.rb

```diff
@@ -452,104 +446,85 @@ def named_entity_data
-        fields << { key: t('site.object.named-entities.where.label'), vals: normalise_named_entity(place[:prefLabel]), multi: true }
-        if !place[:latitude].nil?
-          fields << { key: t('site.object.named-entities.where.latitude'), val: place[:latitude] }
-        end
-        if !place[:longitude].nil?
-          fields << { key: t('site.object.named-entities.where.longitude'), val: place[:longitude] }
-        end
-      end
-      {
-        title: t('site.object.named-entities.where.title'),
-        fields: fields,
-        present: fields.size > 0
-      }
+      named_entity_labels('places', 'where', :latitude, :longitude)
     end
 
     def collect_time_labels
-      fields = []
-      timespans = document.fetch('timespans', [])
-      (timespans).map do |timespan|
-        fields << { key: t('site.object.named-entities.when.term'), val: timespan[:about], agt_link: true }
-        fields << { key: t('site.object.named-entities.when.label'), vals: normalise_named_entity(timespan[:prefLabel]), multi: true }
-        if !timespan[:begin].nil?
-          fields << { key: t('site.object.named-entities.when.begin'), val: timespan[:begin][:def][0] }
-        end
-        if !timespan[:end].nil?
-          fields << { key: t('site.object.named-entities.when.end'), val: timespan[:end][:def][0] }
-        end
-      end
-      {
-        title: t('site.object.named-entities.when.title'),
-        fields: fields,
-        present: fields.size > 0
-      }
+      named_entity_labels('timespans', 'when', :begin, :end)
     end
 
     def collect_concept_labels
-      fields = []
-      concepts = document.fetch('concepts', [])
-      (concepts).map do |concept|
-        fields << { key: t('site.object.named-entities.what.term'), val: concept[:about], agt_link: true }
-        fields << { key: t('site.object.named-entities.what.label'), vals: normalise_named_entity(concept[:prefLabel]), multi: true }
-        broader = concept[:broader]
-        if !broader.nil?
-          multi = broader.size > 1
-          broader = multi ? broader : broader[0]
-          fields << {
-            key: t('site.object.named-entities.what.broader'),
-            val: multi ? nil : broader,
-            vals: multi ? normalise_named_entity(broader, true) : nil,
-            agt_link: true,
-            multi: multi}
+      named_entity_labels('concepts', 'what', :broader)
+    end
+
+    def named_entity_labels(edm, i18n, *args)
+      fields = document.fetch(edm, []).map do |entity|
+        ([:about, :prefLabel] + (args || [])).map do |f|
+          named_entity_field_label(entity, f, i18n)
         end
-      end
+      end.flatten.compact
+
       {
-        title: t('site.object.named-entities.what.title'),
+        title: t("site.object.named-entities.#{i18n}.title"),
         fields: fields,
         present: fields.size > 0
       }
     end
 
+    def named_entity_field_label(entity, field, i18n)
```

Comment:
Cyclomatic complexity for named_entity_field_label is too high. [8/6]<br>Assignment Branch Condition size for named_entity_field_label is too high. [22.34/15]<br>Method has too many lines. [13/10]<br>Perceived complexity for named_entity_field_label is too high. [8/7]